### ISSUE TITLE
Math operation in Binding and x:Bind

### ISSUE THREAD

**Issue by h82258652** [Microsoft/xaml-standard#113]
eg:
```XAML
<FlipView x:Name="ImageFlipView"></FlipView>
<TextBlock Text="{Binding ElementName=ImageFlipView, Path=SelectedIndex+1}"></TextBlock>
<TextBlock Text="{x:Bind ImageFlipView.SelectedIndex+1}"></TextBlock>
```

+, -, *, / and %(mod) should be supported.

**Comment by mossyblog** [comment]
See #102

**Comment by insinfo** [comment]
@h82258652
Something very interesting, I got to implement this in WPF, through custom control

**Comment by mossyblog** [comment]
Last note: I pushed for this to happen in Silverlight 2 .. it was accepted from memory but was moved down in priority due to time issues. It was based off a lot of existing customer feedback. So imho it has merit

**Comment by guibirow** [comment]
Why instead of creating this custom logic inside the view(xaml) does not bind it to an object\property like currently exist, and then create a  converter(Parser\transformer\Formatter)  passing your formula parameters

<TextBox Text="{Binding Something, StringFormat=\{0:P\}}" />

**Comment by h82258652** [comment]
@guibirow 
First, if this xaml code is in ```DataTemplate```, it is very difficult to access them and write the logic code.
Second, converter input parameter type is ```object```, that means boxing (if value is a struct) and performance is pure.
Third, it is hard to write a ```AddConverter```. Here is the code what I implement now.
```C#
public object Convert(object value, Type targetType, object parameter, string language)
{
    var a = (dynamic)value;
    var b = (dynamic)parameter;
    return a + b;
}
```

By the way, ```StringFormat``` is unusable in UWP.

**Comment by kdelorey** [comment]
I don't think this can be implemented as suggested in the `Path` syntax. Tokens like `+` are reserved for parsing casts to nested classes.

For example `{Binding SomeProperty.(SomeClass+NestedClass.Property)}` is casting and accessing the property of a nested class.

**Comment by h82258652** [comment]
@kdelorey But a number is not a vaild class name, no syntax ambiguity here. You can think of it as an speical overload.

**Comment by insinfo** [comment]
@h82258652
I voted for your idea. +1

If possible visit my posts to help in defining the model and syntax of these controls
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/98), 
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/99), 
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/109), 
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/110), 
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/112), 
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/117),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/120),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/121),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/122),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/123),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/124),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/125),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/126),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/127),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/128),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/129),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/130),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/131),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/132),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/135),
[Visit this link](https://github.com/Microsoft/xaml-standard/issues/135),

**Comment by h82258652** [comment]
@insinfo Thanks for vote. I am watching this repo, almost all issues I will read.
